testkit/src/main/scala/org/bitcoins/rpc/BitcoindRpcTestUtil.scala

```diff
@@ -213,62 +243,193 @@ trait BitcoindRpcTestUtil extends BitcoinSLogger {
+      port2: Int = randomPort,
+      rpcPort2: Int = randomPort)(implicit system: ActorSystem): Future[
+    (BitcoindRpcClient, BitcoindRpcClient)] = {
+    implicit val executionContext: ExecutionContext = system.dispatcher
+    val (client1, client2) = createUnconnectedNodePair()
+    client1
+      .addNode(client2.getDaemon.uri, AddNodeArgument.Add)
+      .map(_ => {
+        val try3 = Try(awaitConnection(client1, client2))
+        if (try3.isFailure) {
+          deleteNodePair(client1, client2)
+          throw try3.failed.get
+        }
+      })
+      .map { _ =>
+        client1.generate(100)
+      }
+      .map { _ =>
+        {
+          val try4 = Try(awaitSynced(client1, client2))
+          if (try4.isFailure) {
+            deleteNodePair(client1, client2)
+            throw try4.failed.get
+          }
+          (client1, client2)
+        }
       }
-      client1.generate(100).map { _ =>
-        val try4 = Try(awaitSynced(client1, client2))
+  }
 
-        if (try4.isFailure) {
-          deleteNodePair(client1, client2)
-          throw try4.failed.get
+  def createRawCoinbaseTransaction(
+      sender: BitcoindRpcClient,
+      receiver: BitcoindRpcClient,
+      amount: Bitcoins = Bitcoins(1))(
+      implicit executionContext: ExecutionContext): Future[Transaction] = {
+    sender.generate(2).flatMap { blocks =>
+      sender.getBlock(blocks(0)).flatMap { block0 =>
+        sender.getBlock(blocks(1)).flatMap { block1 =>
+          sender.getTransaction(block0.tx(0)).flatMap { transaction0 =>
+            sender.getTransaction(block1.tx(0)).flatMap { transaction1 =>
+              val input0 =
+                TransactionOutPoint(transaction0.txid.flip,
+                                    UInt32(transaction0.blockindex.get))
+              val input1 =
+                TransactionOutPoint(transaction1.txid.flip,
+                                    UInt32(transaction1.blockindex.get))
+              val sig: ScriptSignature = ScriptSignature.empty
+              receiver.getNewAddress.flatMap { address =>
+                sender.createRawTransaction(
+                  Vector(TransactionInput(input0, sig, UInt32(1)),
+                         TransactionInput(input1, sig, UInt32(2))),
+                  Map(address -> amount))
+              }
+            }
+          }
         }
-        (client1, client2)
       }
     }
   }
 
+  def signRawTransaction(
+      signer: BitcoindRpcClient,
+      transaction: Transaction,
+      utxoDeps: Vector[RpcOpts.SignRawTransactionOutputParameter] = Vector.empty,
+  )(implicit actorMaterializer: ActorMaterializer): Future[
+    SignRawTransactionResult] = signer match {
+    case v17: BitcoindV17RpcClient =>
+      v17.signRawTransactionWithWallet(transaction, utxoDeps)
+    case v16: BitcoindV16RpcClient =>
+      v16.signRawTransaction(transaction, utxoDeps)
+    case unknown: BitcoindRpcClient =>
+      val v16T = BitcoindV16RpcClient.fromUnknownVersion(unknown)
```

Comment:
Why would we need to pattern match try and construct a `V16` or `V17` client when we already know that it isn't one of those. I think this code is uncessary.

Reply:
V16 or V17 clients are only obtained by explicitly requesting one. A general `BitcoindRpcClient` can have both versions in it's underlying `bitcoind`. So `case unknown: BitcoindRpcClient` doesn't mean it's not V16 or V17, it means we're not sure what it is.

We figure out what version we're dealing with by trying to construct V16 and V17 clients, throwing an error if we're unable to construct both of them